purchase_batch_invoicing_advanced/models/purchase_batch_invoicing.py

```diff
@@ -0,0 +1,27 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, _
+from odoo.exceptions import UserError
+
+
+class PurchaseBatchInvoicing(models.TransientModel):
+    _inherit = 'purchase.batch_invoicing'
+
+    def grouped_purchase_orders(self):
+        res = super().grouped_purchase_orders()
+        for record in res:
+            if len(record.mapped('order_type.journal_id')) > 1:
+                raise UserError(_('Selected purchase orders contain different billing journals.'))
+            yield record
+
+    def _prepare_batch_invoice_vals(self, partner):
+        res = super()._prepare_batch_invoice_vals(partner)
+        pos = self.grouped_purchase_orders()
+        for purchase_order in pos:
+            if purchase_order.partner_id == partner:
+                res.update({
+                    'purchase_type_id': purchase_order.order_type.id,
```

Comment:
We've got a problem here: we could get the same journal... with different order types (strange but possible). With this code we'll assign any invoice of the partner the first order type occurrence. Only two solutions are possible:
- prevent grouping, not with different journals, but with different order types (more restrictive).
- enable grouping by order type (very complicated, and not desiderable).

In purchase order tree list, is it available "group by order type" option?

Reply:
`In purchase order tree list, is it available "group by order type" option?`
It is already added by the other addon: https://github.com/OCA/purchase-workflow/blob/c753ef7b1a83f04c05ed748b1b07816af39dd56c/purchase_order_type/views/view_purchase_order.xml#L12

Reply:
OK, then, raising an error if purchases with different type are selected is a good chance, since user can group by this item and easily select purchases with the same type